### ISSUE TITLE
Add Anchor Links to the USWDS 

### ISSUE THREAD

**Issue by mgifford** [uswds/uswds#4166]
**The problem**

Wheither you want to call them heading anchors, linked headings, Anchor Links, or permalinks, it can be especially useful for sites to have an easy means to be able to easily allow folks to link to headings to share information. 

GitHub does it by default when styling markdown files.  
  https://github.com/uswds/uswds#background

The Access Board uses a similar pattern in some places:
  https://www.access-board.gov/ict/#508-chapter-1-application-and-administration

This isn't a problem with short documents, but an increasing number of government pages are very, very long and particularly when referring to legislation or policy it is a real limitation to have to dig through the source code to find an ID to anchor a link against.  

This is useful for both sharing sections of a large document or just wanting to remember where you are on a page if you aren't able to get through it in a single sitting. 

**Describe the solution you'd like**
Couple interesting examples

- https://amberwilson.co.uk/blog/are-your-anchor-links-accessible/
- https://github.com/allejo/jekyll-anchor-headings

**Comment by mgifford** [comment]
Nice to see AnchorJS added here https://beta.ada.gov/topics/intro-to-ada/#other-disability-rights-laws

With so many crazy long government documents having a way to easily link to headers is key. Most folks aren't going to open up dev tools to dig for ID's.

**Comment by mejiaj** [comment]
@mgifford is there any research to validate link at the start vs the end? I like the consistency of having the permalink icon _after_ the header, but visually I like how it looks when it's before the header. I wonder if there are any a11y concerns with having it before the header.

**Comment by mgifford** [comment]
I've done as much research on what has been written on this as I could @mejiaj but couldn't find anything related to best practices about where it is presented.

I find that it is a bit easier to see it in the desktop view when it is to the left of the heading as to the right it seemed to blend in more with the rest of the title. 

Mind you for the mobile view, I prefer having it to the right because we don't want to have that space. We ran into that when we were implementing it here https://accessibility.civicactions.com/

GitHub adds more padding to the left on mobile and leaves the anchor link on the left to allow for that. 

Would love to see some published user research on best practices though!

**Comment by mejiaj** [comment]
Yeah, some research and testing will probably be needed once we get to this.
